File: jdroid-android-firebase-fcm/src/main/java/com/jdroid/android/firebase/fcm/AbstractFcmMessage.java
package com.jdroid.android.firebase.fcm;

import com.google.firebase.messaging.RemoteMessage;

public abstract class AbstractFcmMessage implements FcmMessage {

	public static final String MESSAGE_KEY_EXTRA = "messageKey";

	public abstract String getMessageKey();

	@Override
	public Boolean matches(RemoteMessage remoteMessage) {
		String messageKey = remoteMessage.getData().get(getMessageKeyExtraName());
		return getMessageKey().equalsIgnoreCase(messageKey);
	}

	protected String getMessageKeyExtraName() {
		return MESSAGE_KEY_EXTRA;
	}
}

File: jdroid-android-firebase-fcm/src/main/java/com/jdroid/android/firebase/fcm/AbstractFcmMessage.kt
package com.jdroid.android.firebase.fcm

import com.google.firebase.messaging.RemoteMessage

abstract class AbstractFcmMessage : FcmMessage {

    companion object {
        const val MESSAGE_KEY_EXTRA = "messageKey"
    }

    abstract fun getMessageKey(): String

    override fun matches(remoteMessage: RemoteMessage): Boolean {
        val messageKey = remoteMessage.data[getMessageKeyExtraName()]
        return getMessageKey().equals(messageKey, ignoreCase = true)
    }

    protected fun getMessageKeyExtraName(): String {
        return MESSAGE_KEY_EXTRA
    }
}
